.github/workflows/lint.yml

```diff
@@ -0,0 +1,9 @@
+name: Lint
+on: [ push, pull_request ]
+jobs:
+  lint:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+      - uses: chartboost/ruff-action@v1
```

Comment:
I would to avoid to depends to something not own by the Python Software Foundation or Github. I'm concern by the security it could have. What do you think? 

I propose that we merge with only black for the moment.

Reply:
I've resolved the security concern related to `chartboost/ruff-action@v1` not being a verified action. I've added `lint.sh` to our scripts and now run `lint.sh` directly in the actions, similar to the approach taken in fastapi [1].
---
[1] https://github.com/tiangolo/fastapi/blob/master/.github/workflows/test.yml